jerry-core/ecma/base/ecma-globals.h

```diff
@@ -427,6 +427,18 @@ typedef enum
  */
 #define ECMA_PROPERTY_NAME_TYPE_SHIFT (ECMA_PROPERTY_FLAG_SHIFT + 4)
 
+/**
+ * Convert data property to accessor property.
+ */
+#define ECMA_CONVERT_DATA_PROPERTY_TO_ACCESSOR_PROPERTY(property_p) \
+   *(property_p) = (uint8_t) (*(property_p) + (ECMA_PROPERTY_TYPE_NAMEDACCESSOR - ECMA_PROPERTY_TYPE_NAMEDDATA))
+
+/**
+ * Convert accessor property to data property.
+ */
+#define ECMA_CONVERT_ACCESSOR_PROPERTY_TO_DATA_PROPERTY(property_p) \
+   *(property_p) = (uint8_t) (*(property_p) - (ECMA_PROPERTY_TYPE_NAMEDACCESSOR - ECMA_PROPERTY_TYPE_NAMEDDATA))
```

Comment:
I think you can do both direction with a xor operator:
`*(property_p) ^= ECMA_PROPERTY_TYPE_NAMEDACCESSOR ^ ECMA_PROPERTY_TYPE_NAMEDDATA;`